src/defrag.c

```diff
@@ -1227,6 +1227,7 @@ static void beginDefragCycle(void) {
     defrag.remaining_stages = listCreate();
 
     for (int dbid = 0; dbid < server.dbnum; dbid++) {
+        if (dbHasNoKeys(dbid)) continue;
```

Comment:
I don't think we should skip defrag of empty databases if they exist. The defrag still defrags the internal allocations of the hashtables structs if they exist.

Call chain: defragStageDbKeys -> defragStageKvstoreHelper -> kvstoreHashtableDefragTables -> hashtableDefragTables.

We can skip it if it's uninitialized though (or is it better to do the NULL check inside defragStageDbKeys and defragStageExpiresKvstore instead of here?)

```Suggestion
        if (server.db[dbid] == NULL) continue;
```

Reply:
Good catch. Adding the NULL check here is the right approach.